packages/dev/core/src/Materials/Node/Blocks/Dual/lightBlock.ts

```diff
@@ -29,13 +30,39 @@ import "../../../../Shaders/ShadersInclude/shadowsVertex";
  * Block used to add light in the fragment shader
  */
 export class LightBlock extends NodeMaterialBlock {
-    private _lightId: number;
+    private _lightId: number = 0;
 
     /**
      * Gets or sets the light associated with this block
      */
     public light: Nullable<Light>;
 
+    /** Indicates that no code should be generated in the vertex shader. Can be useful in some specific circumstances (like when doing ray marching for eg) */
+    @editableInPropertyPage("Generate only fragment code", PropertyTypeForEdition.Boolean, "ADVANCED", {
+        notifiers: { rebuild: true, update: true, onValidation: LightBlock._OnGenerateOnlyFragmentCodeChanged },
+    })
+    public generateOnlyFragmentCode = false;
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    private static _OnGenerateOnlyFragmentCodeChanged(block: NodeMaterialBlock, propertyName: string): boolean {
+        const that = block as LightBlock;
+
+        if (that.worldPosition.isConnected) {
+            that.generateOnlyFragmentCode = !that.generateOnlyFragmentCode;
+            console.error("The worldPosition input must not be connected to be able to switch!");

```

Comment:
It would be better if it would be visible in the editor, but I don't know how to (easily) resurface the error up until the editor (`throw` does not work, it also generates the error in the console and not in the status window of the editor)...

One way would be to add an observable in the`NodeMaterialBlock` base class like `onBlockError` and have the editor registers on this observable for all blocks, but that seemed overkill to me, that's why I simply generated the error in the console.